traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go

```diff
@@ -278,14 +300,42 @@ func createV30(w http.ResponseWriter, r *http.Request, inf *api.APIInfo, dsV30 t
 	}
 	return nil, status, userErr, sysErr
 }
+func createV31(w http.ResponseWriter, r *http.Request, inf *api.APIInfo, dsV31 tc.DeliveryServiceV31) (*tc.DeliveryServiceV31, int, error, error) {
+	tx := inf.Tx.Tx
+	dsNullable := tc.DeliveryServiceNullableV30(dsV31)
+	ds := dsNullable.UpgradeToV4()
+	res, status, userErr, sysErr := createV40(w, r, inf, tc.DeliveryServiceV40(ds))
+	if res == nil {
```

Comment:
I'm just doing what the previous versions of this function did (but flipped). It looks like if there is an error in createV40, res will always be nil, otherwise it will have a value.